internal/beater/interceptors/metrics.go

```diff
@@ -99,23 +100,23 @@ func (m *metricsInterceptor) inc(legacyMetricsPrefix string, id request.ResultID
 
 func (m *metricsInterceptor) getCounter(prefix, n string) metric.Int64Counter {
 	name := prefix + n
-	if met, ok := m.counters[name]; ok {
-		return met
+	if met, ok := m.counters.Load(name); ok {
+		return met.(metric.Int64Counter)
 	}
+
 	nm, _ := m.meter.Int64Counter(name)
-	m.counters[name] = nm
-	return nm
+	met, _ := m.counters.LoadOrStore(name, nm)
+	return met.(metric.Int64Counter)
```

Comment:
Great attention to detail. This is correct, as opposed to my bad attempt to fix it in https://github.com/elastic/apm-server/blob/b62fd62bd5f6249f7c06eadbf7fc89f2edc902d0/internal/beater/middleware/monitoring_middleware.go#L76 .